hera_pspec/pstokes.py

```diff
@@ -0,0 +1,166 @@
+   """
+   uv = pyuvdata.UVData()
+   uv.read_miriad(dset)
+   return uv
+
+def _combine_pol(uvd1, uvd2, pol1, pol2, pstokes='pI'):
+   """
+   Reads in miriad file and combines visibilities to form the desired pseudo-stokes visibilities. It returns UVData object containing the pseudo-stokes visibilities
+   
+   Parameters
+   ---------
+   uvd1 : UVData object
+       First UVData object containing data that is used to
+       form Stokes visibilities
+
+   uvd2 : UVData oject
+       Second UVData objects containing data that is used to
+       form Stokes visibilities
+
+   pol1 : Polarization, type: str
+       Polarization of the first UVData object
+  
+   pol2 : Polarization, type: str
+       Polarization of the second UVData object
+
+   pstokes: Pseudo-stokes parameter, type: str
+       Pseudo stokes parameter to form, can be 'pI' or 'pQ' or 'pU' or 'pV'. Default: pI
+  
+   """
+   if isinstance(uvd1, pyuvdata.UVData) == False:
+      raise TypeError("uvd1 must be a pyuvdata.UVData instance")
+   if isinstance(uvd2, pyuvdata.UVData) == False:
+      raise TypeError("uvd2 must be a pyuvdata.UVData instance")
+   
+   assert isinstance(pol1, str)
+   assert isinstance(pol2, str)
+
+   # extracting data array from the UVData objects
+   data1 = uvd1.data_array
+   data2 = uvd2.data_array
+
+   # extracting flag array from the UVdata objects
+   flag1 = uvd1.flag_array
+   flag2 = uvd2.flag_array
+   #constructing flags (boolean)
+   flag = np.logical_or(flag1, flag2)
+   # constructing Stokes visibilities
+   stdata = 0.5 * (pol_weights[pstokes][pol1]*data1 + pol_weights[pstokes][pol2]*data2)
+
+   # assigning and writing data, flags and metadata to UVData object
+   uvdS = copy.deepcopy(uvd1)
+   uvdS.data_array = stdata # pseudo-stokes data
+   uvdS.flag_array = flag # flag array
+   uvdS.polarization_array = np.array([pyuvdata.polstr2num(pstokes)]) # polarization number
+   uvdS.nsample_array = uvd1.nsample_array + uvd2.nsample_array # nsamples
+   uvdS.history = 'merged to form pseudo-stokes visibilities. ' + uvd1.history + uvd2.history # history
+
+   return uvdS
+
+def construct_pstokes(dset1, dset2, pstokes='pI', run_check=True):
+   """
+   Validates datasets required to construct desired visibilities and constructs desired Stokes parameters
+   
+   Parameters
+   ----------
+   dset1 : UVData object or Miriad file
+       First UVData object or Miriad file containing data that is used to
+       form Stokes visibilities
+
+   dset2 : UVData oject or Miriad file
+       Second UVData object or Miriad file containing data that is used to
+       form Stokes visibilities
+
+   pstokes: Stokes parameter, type: str
+       Pseudo stokes parameter to form, can be 'pI' or 'pQ' or 'pU' or 'pV'. Default: I
+
+   run_check: boolean
+      Option to check for the existence and proper shapes of
+      parameters after downselecting data on this object. Default is True.
```

Comment:
This should have a `Returns` block to document what objects are returned by this function.